### ISSUE TITLE
spidermonkey_91 can't be built on i686

### ISSUE THREAD

**Issue by Birdthatcantfly** [NixOS/nixpkgs#177923]
### Steps To Reproduce
1. ``` nix-build '<nixpkgs>' -A pkgsi686Linux.spidermonkey_91 ```

### Build log
```
In file included from Unified_cpp_js_src_jit9.cpp:11:
/build/firefox-91.10.0/js/src/jit/Recover.cpp:312:11: note: 'result' was declared here
  312 |   int32_t result;
      |           ^~~~~~
make[3]: Leaving directory '/build/firefox-91.10.0/obj/js/src'
make[3]: Leaving directory '/build/firefox-91.10.0/obj/js/src/jit'
    Finished release [optimized] target(s) in 2m 25s
/build/firefox-91.10.0/obj/_virtualenvs/common/bin/python -m mozbuild.action.check_binary --target --networking /build/firefox-91.10.0/obj/i686-unknown-linux-gnu/release/libjsrust.a
BUILDTASK {"argv": ["/build/firefox-91.10.0/python/mozbuild/mozbuild/action/check_binary.py", "--target", "--networking", "/build/firefox-91.10.0/obj/i686-unknown-linux-gnu/release/libjsrust.a"], "start": 1655410272.4902613, "end": 1655410272.5648446, "context": null}
make[3]: Leaving directory '/build/firefox-91.10.0/obj/js/src/rust'
make[2]: Leaving directory '/build/firefox-91.10.0/obj'
make[1]: *** [/build/firefox-91.10.0/config/recurse.mk:34: compile] Error 2
make[1]: Leaving directory '/build/firefox-91.10.0/obj'
make: *** [/build/firefox-91.10.0/config/rules.mk:355: default] Error 2
```

### Notify maintainers
@lostnet 

### Metadata
```console
 - system: `"i686-linux"`
 - host os: `Linux 5.15.16-lqx2, NixOS, 22.11 (Raccoon), 22.11pre385606.01e4d1a67f6`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.9.1`
 - channels(root): `"nixos-stable-22.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by lostnet** [comment]
@Birdthatcantfly Can you check if [this pr](https://github.com/NixOS/nixpkgs/pull/172376) has an effect?

**Comment by Birdthatcantfly** [comment]
@lostnet I've got this this error:

```
In file included from Unified_cpp_js_src_jit2.cpp:20:
/build/firefox-91.10.0/js/src/jit/CacheIR.cpp: In member function 'js::jit::AttachDecision js::jit::GetPropIRGenerator::tryAttachPrimitive(js::jit::ValOperandId, JS::HandleId)':
/build/firefox-91.10.0/js/src/jit/CacheIR.cpp:2128:14: warning: 'protoKey' may be used uninitialized in this function [-Wmaybe-uninitialized]
 2128 |   JSProtoKey protoKey;
      |              ^~~~~~~~
make[3]: Leaving directory '/build/firefox-91.10.0/obj/js/src/wasm'
In file included from Unified_cpp_js_src_jit3.cpp:2:
/build/firefox-91.10.0/js/src/jit/CodeGenerator.cpp: In member function 'void js::jit::CodeGenerator::emitIonToWasmCallBase(js::jit::LIonToWasmCallBase<NumDefs>*) [with unsigned int NumDefs = 1]':
/build/firefox-91.10.0/js/src/jit/CodeGenerator.cpp:15441:26: warning: 'argMir' may be used uninitialized in this function [-Wmaybe-uninitialized]
15441 |     ABIArg arg = abi.next(argMir);
      |                  ~~~~~~~~^~~~~~~~
/build/firefox-91.10.0/js/src/jit/CodeGenerator.cpp: In member function 'void js::jit::CodeGenerator::emitIonToWasmCallBase(js::jit::LIonToWasmCallBase<NumDefs>*) [with unsigned int NumDefs = 2]':
/build/firefox-91.10.0/js/src/jit/CodeGenerator.cpp:15441:26: warning: 'argMir' may be used uninitialized in this function [-Wmaybe-uninitialized]
15441 |     ABIArg arg = abi.next(argMir);
      |                  ~~~~~~~~^~~~~~~~
make[3]: Leaving directory '/build/firefox-91.10.0/obj/js/src/jit'
make[2]: Leaving directory '/build/firefox-91.10.0/obj'
make[1]: *** [/build/firefox-91.10.0/config/recurse.mk:34: compile] Error 2
make[1]: Leaving directory '/build/firefox-91.10.0/obj'
make: *** [/build/firefox-91.10.0/config/rules.mk:355: default] Error 2
```

**Comment by OPNA2608** [comment]
Please check the full log of that build and look for `Error 1` (`make` catching the actual error) and any preceding lines with `error:` (actual error from the compiler). What you posted are just harmless warnings and `Error 2` (an error has occurred somewhere else so `make` needs to quit).

**Comment by OPNA2608** [comment]
```console
/nix/store/anc0xd18885xhyp3hrriqn2hq3v3v3ws-gcc-wrapper-10.3.0/bin/g++ -std=gnu++17 -o e_acos.o -c  -I/build/firefox-91.10.0/obj/dist/sy>
modules/fdlibm/src/e_acosh.o
In file included from /build/firefox-91.10.0/modules/fdlibm/src/e_acos.cpp:44:
/build/firefox-91.10.0/modules/fdlibm/src/math_private.h:34:21: error: conflicting declaration 'typedef __double_t double_t'
   34 | typedef __double_t  double_t;
      |                     ^~~~~~~~
In file included from /nix/store/vwggfsx0bjzzdd6bp2phyl342xf01jij-gcc-10.3.0/include/c++/10.3.0/cmath:45,
                 from /build/firefox-91.10.0/obj/dist/system_wrappers/cmath:3,
                 from /build/firefox-91.10.0/modules/fdlibm/src/e_acos.cpp:41:
/nix/store/1xq789b12h7bvc2h64k7vcd71m9jjc7w-glibc-2.34-210-dev/include/math.h:170:21: note: previous declaration as 'typedef long double>
  170 | typedef long double double_t;
      |                     ^~~~~~~~
make[3]: *** [/build/firefox-91.10.0/config/rules.mk:676: e_acos.o] Error 1
```

Upstream issue: https://bugzilla.mozilla.org/show_bug.cgi?id=1729459
